### PR TITLE
Add HoC courses to the set of course offerings to show in the quick assign component

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Adds HoC courses to the response from `/course_offerings/quick_assign_course_offerings`. This object looks a little different from the grade level objects, so I created a separate function to group them, which I plan to reuse for the PL offerings (coming soon!).

Preview of what the new data looks like (not including the grade level data): https://gist.github.com/bethanyaconnor/6676ca6416ab1a9422d564ac2644008d

### CODE REVIEW

File: dashboard/lib/quick_assign_helper.rb [hunk 1]
```diff
@@ -39,4 +41,30 @@ def self.group_offerings(course_offerings, user, locale)
...
+  def self.compare_headers(h1, h2)
+    return -1 if h1 == Curriculum::SharedCourseConstants::COURSE_OFFERING_HEADERS.favorites
+    return 1 if h2 == Curriculum::SharedCourseConstants::COURSE_OFFERING_HEADERS.favorites
+    h1 <=> h2
+  end
+
+  # We want to organize the course offerings by their headers then sort those headers
+  # according to the logic in compare_headers above
+  # The offerings within a header will be sorted alphabetically by display_name.
+  # Any course_offerings that do not a header will be ignored.
```
Comment (hannahbergam) [inline dashboard/lib/quick_assign_helper.rb:55]: tiny nit: 'that do not *have* a header'

Comment (hannahbergam) [inline dashboard/lib/quick_assign_helper.rb]: tiny nit: 'that do not *have a header'

File: dashboard/test/lib/quick_assign_helper.rb [hunk 2]
```diff
@@ -38,25 +56,47 @@ class QuickAssignHelperTest < ActionController::TestCase
...
+
+    assert_equal 1, grouped_offerings['A Header'].length
+    assert_equal course_offering1.id, grouped_offerings['A Header'][0][:id]
+    assert_equal 2, grouped_offerings['B Header'].length
+    assert_equal course_offering3.id, grouped_offerings['B Header'][0][:id]
+    assert_equal course_offering2.id, grouped_offerings['B Header'][1][:id]
+    assert_equal 1, grouped_offerings['Favorites'].length
+    assert_equal favorite_course_offering.id, grouped_offerings['Favorites'][0][:id]
+
+    assert_equal ['Favorites', 'A Header', 'B Header'], grouped_offerings.keys
```
Comment (hannahbergam) [inline dashboard/test/lib/quick_assign_helper.rb:100]: Thank you for testing this!

**Review by hannahbergam — APPROVED** [review]
LGTM! Tbh the sorting makes my brain hurt but I trust the tests:)

**Comment by bethanyaconnor** [comment]
> the sorting makes my brain hurt but I trust the tests

This makes me more glad that I put this complex business logic into a separate file! Hopefully, once we're done with this project, this will just live forever out of site and not cause too many people headers 😀

File: dashboard/test/lib/quick_assign_helper.rb (hunk 2 above)
Comment (bethanyaconnor) [reply dashboard/test/lib/quick_assign_helper.rb:100]: I'm glad I did because it was way harder than I expected 🙃

**Pull request merged**